### ISSUE TITLE
[YSQL] Slowdown single insert 10x if once used bulk insert(transaction)

### ISSUE THREAD

**Issue by adrchw** [YugaByte/yugabyte-db#1570]
Hi, We discovered BUG when we are testing inserts. If once we run bulk insert(transaction) we slowdown all single inserts for the same table permanently by about 5-10 times.

Tested in 2 environments:
1. Dell XPS 15 (8x 2.6 GHz Intel Core i7, 16GB RAM, Linux Mint 18.1)
1. Google Cloud Cluster (3 nodes with 2 vCPUs and 3 GB memory, Debian 9)

### Way to reproduce
1) Run the script with **$single = true;**
2) Run the script with **$single = false;**
3) Run the script with **$single = true;**

After step 3 you can see a big drop in inserts/sec/

### Test result:

```
taken: 60.000118017197 | items: 79718 | Ops/sec: 1329 | SINGLE
taken: 60.042438030243 | items: 75800 | Ops/sec: 1262
taken: 60.003437995911 | items: 7736 | Ops/sec: 129 | SINGLE
```

### Schema:
```
CREATE TABLE t_v2 (
                                    user_id UUID NOT NULL,
                                    rand_int integer,
                                    md5id varchar(32) NULL,
                                    random varchar(2000) NULL,
                                    date_added TIMESTAMPTZ NULL,
                                    PRIMARY KEY (rand_int, user_id)
);
```

### Code: 
```
$dbh = new PDO('pgsql:host=localhost;port=5433;dbname=postgres',
		'postgres', null, array(
				PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION,
				PDO::ATTR_EMULATE_PREPARES => true,
				PDO::ATTR_PERSISTENT => true
		));

$start = microtime(true);
$i=0;
$single = true;
$time = 60;

if($single)
{
	while(true)
	{
		$dbh->prepare('INSERT INTO t_v2 (user_id, rand_int, md5id,random,date_added) VALUES (?, ?, ?, ?, ?)')
				->execute([
						(string)new \Cassandra\Uuid(),
						rand(0,10),
						md5($uid = uniqid("a", true)),
						$uid,
						date(DATE_ATOM)
				]);

		$i++;
		if(microtime(true)-$start > $time) break;
	}
}
else
{
	$bulkQuery = [];
	$bulkQueryVal = [];
	while(true)
	{
		$bulkQuery[] = '(?, ?, ?, ?, ?)';
		$bulkQueryVal = array_merge($bulkQueryVal, array_values([
				(string)new \Cassandra\Uuid(),
				rand(0,10),
				md5($uid = uniqid("a", true)),
				$uid,
				date(DATE_ATOM)
		]));
		$i++;

		if($i % 200 == 0)
		{
			$dbh->prepare('INSERT INTO t_v2 (user_id, rand_int, md5id,random,date_added) VALUES '.implode(',', $bulkQuery))
					->execute($bulkQueryVal);

			$bulkQuery = [];
			$bulkQueryVal = [];
		}

		if(microtime(true)-$start > $time) break;
	}

	if($bulkQuery)
	{
		$dbh->prepare('INSERT INTO t_v2 (user_id,rand_int, md5id,random,date_added) VALUES '.implode(',', $bulkQuery))
				->execute($bulkQueryVal);
	}
}

$taken = microtime(true)-$start;
echo "taken: $taken | items: $i | Ops/sec: ".round($i/$taken).(@$single?' | SINGLE':'')."\n";
```
--
Adrian Chwialkowski
Feedoptimise

**Comment by kmuthukk** [comment]
Thx a lot for reporting this issue @adrchw and providing all the details. 

As discussed offline, we are also able to repro this issue, and will try to get to the bottom of it soon.

**Comment by kmuthukk** [comment]
@adrchw - this appears to be a variant of the hot row problem (described in #1240) that we are working on optimizing.

In your test case, even though, each row has a unique PK `(rand_int, user_id)`,  the program uses `rand(0, 10)` for the rand_int column and a guid for the user_id. It has only 10 distinct values for `rand_int`, and this triggers a variant of the hot row problem described in https://github.com/YugaByte/yugabyte-db/issues/1240 (essentially, it is a hot common prefix of the primary key).

Interim suggestions to avoid issue:

* flip the PK column order to `user_id, rand_int` from `rand_int, user_id` 

* use a wider set of random ints, say`rand(0, 1000000)`

**Comment by kmuthukk** [comment]
Was able to confirm that this problem has been fixed in 1.3.1 (with #1240 fix).
